client/python/openlineage/client/transport/factory.py

```diff
@@ -68,4 +70,7 @@ def _create_transport(self, config: dict[str, str]) -> Transport:
             msg = f"Config {config_class} has to be class, and subclass of Config"
             raise TypeError(msg)
 
-        return transport_class(config_class.from_dict(config))  # type: ignore[call-arg]
+        transport: Transport = transport_class(config_class.from_dict(config))  # type: ignore[call-arg]
+        if transport_name:
```

Comment:
```suggestion
        if transport_name and not getattr(transport, "name"):
```
Right now we are overwriting whatever was there. Probably nobody had attribute named `name` it in their custom transport, but maybe we should check?

Reply:
Not sure why use `getattr` since all transports need to inherit from `openlineage.client.transport.Transport` that now would have `name` property added.
Still, I'm adding the condition as suggested.